// @lmig/claim-ui-components/dist/expander/expander.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class Expander extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { isExpanded: props.isExpanded };
    this.toggleExpand = this.toggleExpand.bind(this);
    this.setStyles = this.setStyles.bind(this);
  }

  getChildContext() {
    return {
      isExpanded: this.state.isExpanded,
      toggleExpand: this.toggleExpand,
      setStyles: this.setStyles,
      bodyStyle: this.state.isExpanded ? this.state.bodyExpandedStyle : this.state.bodyCollapsedStyle
    };
  }

  toggleExpand() {
    this.setState({ isExpanded: !this.state.isExpanded });
  }

  setStyles({ offsetHeight }) {
    const bodyExpandedStyle = { height: offsetHeight };
    const bodyCollapsedStyle = { height: 0 };
    this.setState({ bodyExpandedStyle, bodyCollapsedStyle });
  }

  render() {
    return _react2.default.createElement(
      'div',
      { className: `expander ${this.props.className}` },
      this.props.children
    );
  }
}

exports.default = Expander;
Expander.defaultProps = {
  isExpanded: false,
  className: ''
};

Expander.propTypes = {
  isExpanded: _propTypes2.default.bool,
  className: _propTypes2.default.string,
  children: _propTypes2.default.node
};

Expander.childContextTypes = {
  isExpanded: _propTypes2.default.bool,
  toggleExpand: _propTypes2.default.func,
  setStyles: _propTypes2.default.func,
  bodyStyle: _propTypes2.default.object
};
