// @lmig/claim-ui-components/dist/phase-card/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PhaseCardWhen = exports.PhaseCardText = exports.PhaseCardSubheading = exports.PhaseCardDescription = exports.PhaseCardDetail = exports.PhaseCardDetailList = exports.PhaseCardHeading = exports.PhaseCard = undefined;

var _phaseCard = require('./phase-card');

var _phaseCard2 = _interopRequireDefault(_phaseCard);

var _phaseCardHeading = require('./phase-card-heading');

var _phaseCardHeading2 = _interopRequireDefault(_phaseCardHeading);

var _phaseCardDetailList = require('./phase-card-detail-list');

var _phaseCardDetailList2 = _interopRequireDefault(_phaseCardDetailList);

var _phaseCardDetail = require('./phase-card-detail');

var _phaseCardDetail2 = _interopRequireDefault(_phaseCardDetail);

var _phaseCardDescription = require('./phase-card-description');

var _phaseCardDescription2 = _interopRequireDefault(_phaseCardDescription);

var _phaseCardSubheading = require('./phase-card-subheading');

var _phaseCardSubheading2 = _interopRequireDefault(_phaseCardSubheading);

var _phaseCardText = require('./phase-card-text');

var _phaseCardText2 = _interopRequireDefault(_phaseCardText);

var _phaseCardWhen = require('./phase-card-when');

var _phaseCardWhen2 = _interopRequireDefault(_phaseCardWhen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.PhaseCard = _phaseCard2.default;
exports.PhaseCardHeading = _phaseCardHeading2.default;
exports.PhaseCardDetailList = _phaseCardDetailList2.default;
exports.PhaseCardDetail = _phaseCardDetail2.default;
exports.PhaseCardDescription = _phaseCardDescription2.default;
exports.PhaseCardSubheading = _phaseCardSubheading2.default;
exports.PhaseCardText = _phaseCardText2.default;
exports.PhaseCardWhen = _phaseCardWhen2.default;
