// @lmig/claim-ui-components/dist/phase-card/phase-card.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class PhaseCard extends _react.Component {
  getChildContext() {
    return {
      phaseState: this.props.phaseState
    };
  }

  render() {
    var _props = this.props;
    const children = _props.children,
          className = _props.className,
          phaseState = _props.phaseState,
          phaseStateIconMap = _props.phaseStateIconMap;


    return _react2.default.createElement(
      'div',
      { className: `phase-card ${className}` },
      _react2.default.createElement(
        'div',
        { className: `phase-card-status ${phaseState}` },
        _react2.default.createElement('span', { className: `phase-card-icon ${phaseStateIconMap[phaseState]}` })
      ),
      _react2.default.createElement(
        'div',
        { className: 'phase-card-body' },
        children
      )
    );
  }
}

exports.default = PhaseCard;
PhaseCard.defaultProps = {
  className: '',
  phaseState: '',
  phaseStateIconMap: {}
};

PhaseCard.propTypes = {
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  classNameIcon: _propTypes2.default.string,
  phaseState: _propTypes2.default.string,
  phaseStateIconMap: _propTypes2.default.object
};

PhaseCard.childContextTypes = {
  phaseState: _propTypes2.default.string
};
